kgforge/specializations/models/rdf/service.py

```diff
@@ -128,21 +140,34 @@ def lookup_shape_from_node(self, node: URIRef) -> Shape:
 
 class RdfService:
 
-    def __init__(self, graph: Graph, context_iri: Optional[str] = None) -> None:
+    def __init__(
+        self, graph: rdflib.Dataset, context_iri: Optional[str] = None
+    ) -> None:
 
         if context_iri is None:
             raise ConfigurationError("RdfModel requires a context")
         self._graph = graph
+        self.NXV = Namespace("https://bluebrain.github.io/nexus/vocabulary/")
         self._context_cache = {}
-        self.classes_to_shapes = self._build_shapes_map()
         resolved_context = self.resolve_context(context_iri)
         self.context = Context(resolved_context, context_iri)
-        self.types_to_shapes: Dict = self._build_types_to_shapes()
+        (
+            self.class_to_shape,
+            self.shape_to_defining_resource,
+            self.shape_to_named_graph,
+            self.defining_resource_to_named_graph,
+        ) = self._build_shapes_map()
+        self._imported = []
 
-    def schema_source_id(self, schema_iri: str) -> str:
+    @abstractmethod
+    def schema_source_id(self, shape_uri: str) -> str:
```

Comment:
I think this shouldn't be abstract and this should have 
return self.shape_to_defining_resource[schema_iri]
Which is what both implementations currently have.
I don't see why any additional logic could be envisioned that would justify having the subclasses implement it

Reply:
schema_source_id has different implementation in the services. schema_id() is the one with the same implementation currrently

Reply:
I wonder why though, I would think in the case of loading the model from Nexus, the named graph and the defining resource would be the same thing